From: Phil Elwell <phil@raspberrypi.com>
Date: Fri, 17 Apr 2020 16:20:55 +0100
Subject: Revert "firmware: raspberrypi: register clk device"

This reverts commit 91f2cf4a6b2131016b1ae9c9500245f0572112c7.

Revert this because the clock driver won't probe without a DT node, and
creating it as a platform device won't give it one. Doing so avoids the
following error message:

    raspberrypi-clk raspberrypi-clk: Missing firmware node

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 drivers/firmware/raspberrypi.c | 10 ----------
 1 file changed, 10 deletions(-)

diff --git a/drivers/firmware/raspberrypi.c b/drivers/firmware/raspberrypi.c
index 4f3ab14d275a..dc11a1968e99 100644
--- a/drivers/firmware/raspberrypi.c
+++ b/drivers/firmware/raspberrypi.c
@@ -21,7 +21,6 @@
 #define MBOX_CHAN_PROPERTY		8
 
 static struct platform_device *rpi_hwmon;
-static struct platform_device *rpi_clk;
 
 struct rpi_firmware {
 	struct mbox_client cl;
@@ -299,12 +298,6 @@ rpi_register_hwmon_driver(struct device *dev, struct rpi_firmware *fw)
 	}
 }
 
-static void rpi_register_clk_driver(struct device *dev)
-{
-	rpi_clk = platform_device_register_data(dev, "raspberrypi-clk",
-						-1, NULL, 0);
-}
-
 static int rpi_firmware_probe(struct platform_device *pdev)
 {
 	struct device *dev = &pdev->dev;
@@ -334,7 +327,6 @@ static int rpi_firmware_probe(struct platform_device *pdev)
 	rpi_firmware_print_firmware_revision(fw);
 	rpi_firmware_print_firmware_hash(fw);
 	rpi_register_hwmon_driver(dev, fw);
-	rpi_register_clk_driver(dev);
 
 	return 0;
 }
@@ -355,8 +347,6 @@ static int rpi_firmware_remove(struct platform_device *pdev)
 
 	platform_device_unregister(rpi_hwmon);
 	rpi_hwmon = NULL;
-	platform_device_unregister(rpi_clk);
-	rpi_clk = NULL;
 	mbox_free_channel(fw->chan);
 	g_pdev = NULL;
 
